lochness/redcap/__init__.py

```diff
@@ -680,13 +680,27 @@ def sync(Lochness, subject, dry=False):
                 # UPENN REDCap is set up with its own record_id, but have added
                 # "session_subid" field to note AMP-SCZ ID
                 redcap_subject_sl = redcap_subject.lower()
+                digits = [1, 2, 3, 4, 5, 6, 7, 8, 9]
+                digits_str = [str(x) for x in digits]
+                contains_logic = []
+                for subject_id in [redcap_subject, redcap_subject_sl]:
+                    contains_logic += [
+                            f"contains([session_subid], '{subject_id}_{x}')"
+                            for x in digits_str]
+                    contains_logic += [
+                            f"contains([session_subid], '{subject_id}={x}')"
+                            for x in digits_str]
+
+
                 record_query = {
                     'token': api_key,
                     'content': 'record',
                     'format': 'json',
                     'filterLogic': f"[session_subid] = '{redcap_subject}' or "
                                    f"[session_subid] = '{redcap_subject_sl}'"
                 }
```

Comment:
Shouldn't the `record_query` be pushed inside the `for` loop?

Reply:
In its current setting, you seem to be querying REDCap for all possible IDs for a subject in one single query in line 717. Does REDCap work like that?

Reply:
The current design creates `contains_logic` in the loop, then will be used in the record_query once. So, no need to include `record_query` inside the loop.

Yes, the `contains` logic can include multiple logics in the single call. Notice that I'm not including this `contains_logic` yet to `record_query`, because we're waiting for Sylvain's confirmation on whether to include `=n` pattern as well as `_n` pattern.

Reply:
I'll commit more once we know more about the SOP to follow.